crates/revm/src/db/in_memory_db.rs

```diff
@@ -87,26 +87,12 @@ impl<ExtDB: DatabaseRef> CacheDB<ExtDB> {
 impl<ExtDB: DatabaseRef> DatabaseCommit for CacheDB<ExtDB> {
     fn commit(&mut self, changes: Map<H160, Account>) {
         for (add, acc) in changes {
-            if acc.is_empty() || matches!(acc.filth, Filth::Destroyed) {
-                self.cache.remove(&add);
-                self.storage.remove(&add);
```

Comment:
Here, you need to set cache to `Account::default` and iterate over `storage.get(add).iter()` and set them all to zero

Reply:
@rakita you said set "cache to Account::default" but we're currently setting the `changes`? is that right, or a mistake?

Reply:
@gakonst `changes` are added after, because of this: https://github.com/bluealloy/revm/pull/126#issuecomment-1152717104